apps/src/code-studio/pd/application_dashboard/detail_view_application_specific_questions.jsx

```diff
@@ -60,34 +64,61 @@ export default class DetailViewApplicationSpecificQuestions extends React.Compon
     return questionNumber + questionText;
   };
 
+  handlePrincipalApprovalChange = (_id, principalApproval) => {
+    this.setState({principalApproval});
+  };
+
+  renderNoPrincipalApprovalMessage() {
```

Comment:
It's a little counterintuitive to have the word message in here - it's a button not a message in one case

Reply:
Good point. I'll rename `renderNoPrincipalApprovalButtons`